Sources/MapboxCoreNavigation/HistoryEvent.swift

```diff
@@ -49,14 +49,21 @@ internal class StatusUpdateHistoryEvent: UnknownHistoryEvent {
     }
 }
 
-internal class PushRecordHistoryEvent: UnknownHistoryEvent {
-    let type: String
-    let properties: String
+/// History event being pushed by the user
+///
+/// Such events are created by calling `HistoryRecording.pushHistoryEvent(type:jsonData:)`.
+public struct UserPushedHistoryEvent: HistoryEvent {
```

Comment:
Code comment explains how such events are created, so I wanted to retain some similarities in naming with `HistoryRecording.pushHistoryEvent(type:jsonData:)`. I don't think it may be confuse developers into thinking that app users actually putting something directly into history file